jerry-core/api/jerry.c

```diff
@@ -4624,15 +4624,15 @@ jerry_value_t
 jerry_get_new_target (void)
 {
 #if ENABLED (JERRY_ESNEXT)
-  ecma_object_t *current_new_target = JERRY_CONTEXT (current_new_target);
+  ecma_object_t *current_new_target_p = JERRY_CONTEXT (current_new_target);
```

Comment:
Could you please check the other mispellings as well? Also, I's suggest to rename `JERRY_CONTEXT (current_new_target)` to `JERRY_CONTEXT (current_new_target_p)`

Reply:
80 results - 26 files

jerry-core\api\jerry.c:
  2069  #if ENABLED (JERRY_BUILTIN_PROMISE)
  2070:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  2071  

  2073    {
  2074:     JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ECMA_BUILTIN_ID_PROMISE);
  2075    }

  2078  
  2079:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  2080    return promise_value;

  4626  #if ENABLED (JERRY_ESNEXT)
  4627:   ecma_object_t *current_new_target = JERRY_CONTEXT (current_new_target);
  4628  
  4629:   if (current_new_target == NULL)
  4630    {

  4633  
  4634:   ecma_ref_object (current_new_target);
  4635:   return ecma_make_object_value (current_new_target);
  4636  #else /* !ENABLED (JERRY_ESNEXT) */

  5095    };
  5096:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  5097    if (old_new_target_p == NULL)
  5098    {
  5099:     JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ECMA_BUILTIN_ID_DATAVIEW);
  5100    }

  5102    ecma_value_t dataview_value = ecma_op_dataview_create (arguments_p, 3);
  5103:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  5104    return jerry_return (dataview_value);

  5661    }
  5662:   ecma_object_t * old_new_target_p = JERRY_CONTEXT (current_new_target);
  5663  

  5665    {
  5666:     JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ctor_id);
  5667    }

  5673  
  5674:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  5675    return container_value;

jerry-core\ecma\base\ecma-init-finalize.c:
  63  #if ENABLED (JERRY_ESNEXT)
  64:   JERRY_CONTEXT (current_new_target) = NULL;
  65  #endif /* ENABLED (JERRY_ESNEXT) */

  74  #if ENABLED (JERRY_ESNEXT)
  75:   JERRY_ASSERT (JERRY_CONTEXT (current_new_target) == NULL);
  76  #endif /* ENABLED (JERRY_ESNEXT) */

jerry-core\ecma\builtin-objects\ecma-builtin-array.c:
  516  #else /* ENABLED (JERRY_ESNEXT) */
  517:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  518                                                                     ECMA_BUILTIN_ID_ARRAY_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-date.c:
  721  #if ENABLED (JERRY_ESNEXT)
  722:   JERRY_ASSERT (JERRY_CONTEXT (current_new_target));
  723  
  724:   ecma_object_t *prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  725                                                                             ECMA_BUILTIN_ID_DATE_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-error.c:
  69  #else /* ENABLED (JERRY_ESNEXT) */
  70:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  71                                                                     ECMA_BUILTIN_ID_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-evalerror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_EVAL_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-object.c:
  126  #if ENABLED (JERRY_ESNEXT)
  127:   if (JERRY_CONTEXT (current_new_target) != ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT))
  128    {
  129:     ecma_object_t *prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  130                                                                               ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-rangeerror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_RANGE_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-referenceerror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_REFERENCE_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-regexp.c:
  53    bool free_arguments = false;
  54:   ecma_object_t *new_target_p = JERRY_CONTEXT (current_new_target);
  55  #else /* !ENABLED (JERRY_ESNEXT) */

jerry-core\ecma\builtin-objects\ecma-builtin-syntaxerror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_SYNTAX_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-typeerror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_TYPE_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\ecma-builtin-urierror.c:
  71  #else /* ENABLED (JERRY_ESNEXT) */
  72:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  73                                                                     ECMA_BUILTIN_ID_URI_ERROR_PROTOTYPE);

jerry-core\ecma\builtin-objects\typedarray\ecma-builtin-typedarray-helpers.c:
  43    ecma_builtin_id_t proto_id = ecma_typedarray_helper_get_prototype_id (typedarray_id);
  44:   ecma_object_t *current_new_target_p = JERRY_CONTEXT (current_new_target);
  45    ecma_object_t *prototype_obj_p;
  46  
  47:   if (current_new_target_p != NULL)
  48    {
  49:     prototype_obj_p = ecma_op_get_prototype_from_constructor (current_new_target_p, proto_id);
  50      if (prototype_obj_p == NULL)

  65  
  66:   if (current_new_target_p != NULL)
  67    {

jerry-core\ecma\operations\ecma-arraybuffer-object.c:
  145  
  146:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  147                                                                     ECMA_BUILTIN_ID_ARRAYBUFFER_PROTOTYPE);

jerry-core\ecma\operations\ecma-async-generator-object.c:
  57  
  58:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  59:   JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ECMA_BUILTIN_ID_PROMISE);
  60    ecma_value_t result = ecma_op_create_promise_object (ECMA_VALUE_EMPTY, ECMA_PROMISE_EXECUTOR_EMPTY);
  61:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  62    task_p->promise = result;

jerry-core\ecma\operations\ecma-boolean-object.c:
  57  #if ENABLED (JERRY_ESNEXT)
  58:   ecma_object_t *new_target = JERRY_CONTEXT (current_new_target);
  59    if (new_target)

jerry-core\ecma\operations\ecma-container-object.c:
  375                  || lit_id == LIT_MAGIC_STRING_WEAKSET_UL);
  376:   JERRY_ASSERT (JERRY_CONTEXT (current_new_target) != NULL);
  377  
  378:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target), proto_id);
  379  

jerry-core\ecma\operations\ecma-dataview-object.c:
   50    JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
   51:   JERRY_ASSERT (JERRY_CONTEXT (current_new_target));
   52  

  123    /* 9. */
  124:   ecma_object_t *prototype_obj_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  125                                                                             ECMA_BUILTIN_ID_DATAVIEW_PROTOTYPE);

jerry-core\ecma\operations\ecma-function-object.c:
   493  #if ENABLED (JERRY_ESNEXT)
   494:   ecma_object_t *new_target_p = JERRY_CONTEXT (current_new_target);
   495  

   651  
   652:   if (JERRY_CONTEXT (current_new_target) != NULL)
   653    {
   654:     arrow_func_p->new_target = ecma_make_object_value (JERRY_CONTEXT (current_new_target));
   655    }

  1108      {
  1109:       JERRY_CONTEXT (current_new_target) = NULL;
  1110      }

  1112      {
  1113:       JERRY_CONTEXT (current_new_target) = ecma_get_object_from_value (arrow_func_p->new_target);
  1114      }

  1158    {
  1159:     if (JERRY_CONTEXT (current_new_target) == NULL)
  1160      {

  1400  #if ENABLED (JERRY_ESNEXT)
  1401:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  1402    if (JERRY_UNLIKELY (!(JERRY_CONTEXT (status_flags) & ECMA_STATUS_DIRECT_EVAL)))
  1403    {
  1404:     JERRY_CONTEXT (current_new_target) = NULL;
  1405    }

  1423  #if ENABLED (JERRY_ESNEXT)
  1424:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  1425  #endif /* ENABLED (JERRY_ESNEXT) */

  1491  #if ENABLED (JERRY_ESNEXT)
  1492:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  1493:   JERRY_CONTEXT (current_new_target) = new_target_p;
  1494  #endif /* ENABLED (JERRY_ESNEXT) */

  1498  #if ENABLED (JERRY_ESNEXT)
  1499:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  1500  #endif /* ENABLED (JERRY_ESNEXT) */

  1557  #if ENABLED (JERRY_ESNEXT)
  1558:       ecma_object_t *old_new_target = JERRY_CONTEXT (current_new_target);
  1559:       JERRY_CONTEXT (current_new_target) = new_target_p;
  1560  #endif /* ENABLED (JERRY_ESNEXT) */

  1564  #if ENABLED (JERRY_ESNEXT)
  1565:       JERRY_CONTEXT (current_new_target) = old_new_target;
  1566  #endif /* ENABLED (JERRY_ESNEXT) */

  1609    /* 6. */
  1610:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  1611:   JERRY_CONTEXT (current_new_target) = new_target_p;
  1612  #endif /* ENABLED (JERRY_ESNEXT) */

  1616  #if ENABLED (JERRY_ESNEXT)
  1617:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
  1618  #endif /* ENABLED (JERRY_ESNEXT) */

jerry-core\ecma\operations\ecma-number-object.c:
  62  #if ENABLED (JERRY_ESNEXT)
  63:   ecma_object_t *new_target = JERRY_CONTEXT (current_new_target);
  64    if (new_target)

jerry-core\ecma\operations\ecma-promise-object.c:
  392  {
  393:   JERRY_ASSERT (JERRY_CONTEXT (current_new_target) != NULL);
  394    /* 3. */
  395:   ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
  396                                                                     ECMA_BUILTIN_ID_PROMISE_PROTOTYPE);

jerry-core\ecma\operations\ecma-string-object.c:
  72  #if ENABLED (JERRY_ESNEXT)
  73:   ecma_object_t *new_target = JERRY_CONTEXT (current_new_target);
  74    if (new_target)

jerry-core\jcontext\jcontext.h:
  230     */
  231:   ecma_object_t *current_new_target;
  232  #endif /* ENABLED (JERRY_ESNEXT) */

jerry-core\vm\opcodes.c:
   850    /* inside the generators the "new.target" is always "undefined" as it can't be invoked with "new" */
   851:   ecma_object_t *old_new_target = JERRY_CONTEXT (current_new_target);
   852:   JERRY_CONTEXT (current_new_target) = NULL;
   853  

   864  
   865:   JERRY_CONTEXT (current_new_target) = old_new_target;
   866    executable_object_p->extended_object.u.class_prop.extra_info &= (uint16_t) ~ECMA_EXECUTABLE_OBJECT_RUNNING;

   972  
   973:   ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
   974:   JERRY_CONTEXT (current_new_target) = promise_p;
   975  

   980  
   981:   JERRY_CONTEXT (current_new_target) = old_new_target_p;
   982    return result;

  1140  
  1141:   if (JERRY_CONTEXT (current_new_target) == NULL)
  1142    {

  1164  
  1165:   if (JERRY_CONTEXT (current_new_target) == NULL)
  1166    {

  1180    ecma_value_t result = ecma_op_function_construct (super_ctor_p,
  1181:                                                     JERRY_CONTEXT (current_new_target),
  1182                                                      args_p,

jerry-core\vm\vm.c:
   604      completion_value = ecma_op_function_construct (func_obj_p,
   605:                                                    JERRY_CONTEXT (current_new_target),
   606                                                     arguments_p,

  2628            {
  2629:             ecma_object_t *old_new_target_p = JERRY_CONTEXT (current_new_target);
  2630:             JERRY_CONTEXT (current_new_target) = ecma_builtin_get (ECMA_BUILTIN_ID_PROMISE);
  2631  

  2633  
  2634:             JERRY_CONTEXT (current_new_target) = old_new_target_p;
  2635            }

  2700          {
  2701:           ecma_object_t *new_target_object = JERRY_CONTEXT (current_new_target);
  2702            if (new_target_object == NULL)

Do we need rename them all?